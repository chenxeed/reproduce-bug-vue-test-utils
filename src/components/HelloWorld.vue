<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <div>
      Modal Link<br>
      <modal-link v-model="url" />
      The URL : {{ url }}
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import ModalLink from './ModalLink.vue';

@Component({
  components : {
    ModalLink
  }
})
export default class HelloWorld extends Vue {
  @Prop() private msg!: string;

  url: string = "";
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
