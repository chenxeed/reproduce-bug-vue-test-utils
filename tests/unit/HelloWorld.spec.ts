import { shallowMount, mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'

describe('HelloWorld.vue', () => {
  it('render', () => {
    const wrapper = shallowMount(HelloWorld)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders props.msg when passed', () => {
    const msg = 'new message'
    const wrapper = shallowMount(HelloWorld, {
      propsData: { msg }
    })
    expect(wrapper.text()).toMatch(msg)
  })

  it("connects data 'url' with child component ModalLink (mount)", () => {
    const wrapper = mount(HelloWorld)
    const modalLink = wrapper.find({ name: 'ModalLink' })
    wrapper.setData({ url : 'somethingelse' })
    expect( modalLink.vm.url ).toBe( 'somethingelse' )
  })

  it("connects data 'url' with child component ModalLink (shallowMount)", () => {
    const wrapper = shallowMount(HelloWorld)
    const modalLink = wrapper.find({ name: 'ModalLink' })
    wrapper.setData({ url : 'somethingelse' })
    expect( modalLink.vm.url ).toBe( 'somethingelse' )
  })
})
