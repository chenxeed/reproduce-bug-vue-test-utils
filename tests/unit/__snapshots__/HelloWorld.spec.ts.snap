// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue render 1`] = `
<div class="hello">
  <h1></h1>
  <div>
    Modal Link<br>
    <modallink-stub></modallink-stub>
    The URL :
  </div>
</div>
`;
